pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/ASTMethod.java

```diff
@@ -24,17 +36,30 @@ public Object jjtAccept(ApexParserVisitor visitor, Object data) {
         return visitor.visit(this, data);
     }
 
+    /**
+     * Returns the name of the method, converting the parser name to the internal PMD name as
+     * needed.
+     */
+    private String getName() {
+        if (node.isAnonymousInitializationCode()) {
+            return STATIC_INIT_ID;
+        } else if (node.isConstructor()) {
+            return CONSTRUCTOR_ID;
+        } else if (getParent() instanceof ASTProperty) {
+            return ASTProperty.formatAccessorName((ASTProperty) getParent());
+        } else {
+            return node.getId().getString();
+        }
+    }
+
     @Override
     public String getImage() {
-        // return node.getMethodInfo().getName();
-        // TODO(b/239648780)
-        return null;
+        return getName();
+        // TODO(b/239648780): differs from #getCanonicalName in some instances
```

Comment:
In case of constructors: `getName()` would be the actual name of the constructor. `getCanonicalName()` would be `<init>`.

Reply:
Thanks, updated. I tested out the other cases handled here to confirm that both `getImage` and `getCanonicalName` match the previous behavior.